torch/_inductor/select_algorithm.py

```diff
@@ -849,47 +846,48 @@ def benchmark_in_current_process(choice):
             torch.cuda.synchronize()  # shake out any CUDA errors
             return result
 
-        def benchmark_in_sub_process(choice):
-            # only benchmark triton kernel in sub process for now.
-            # ATen/Extern kernel are still benchmarked in the current process.
-            if isinstance(choice, ExternKernelCaller):
-                return benchmark_in_current_process(choice)
-
+        def benchmark_in_current_process(options):
+            timings = {}
+            for choice in options:
+                try:
+                    timing = benchmark_choice_in_current_process(choice)
+                except RuntimeError as e:
+                    msg = str(e)
+                    if "invalid argument" in msg:
+                        msg += "\n\nThis may mean this GPU is too small for max_autotune mode.\n\n"
+                        log.warning(msg)
+                        timing = float("inf")
+                    else:
+                        if "illegal memory access" in msg:
+                            msg += "\n\nEither error in template or triton bug.\n"
+                        raise ErrorFromChoice(msg, choice, debug_str())
+                except AssertionError as e:
+                    raise AssertionError(
+                        f"Incorrect result from choice {choice}\n\n{e}"
+                    )
+
+                timings[choice] = timing
+
+            return timings
+
+        def benchmark_in_sub_process(options):
             from . import autotune_process
 
-            if DEBUG:
-                start_ts = time.time()
+            # only benchmark triton kernel in sub process for now.
+            # ATen/Extern kernel are still benchmarked in the current process.
```

Comment:
is there a reason for this ?

Reply:
@eellison I don't know. Maybe @shunting314 can help answer this part too? I maintained this part of the logic from the existing implementation. If there's no real technical reason why we can't use the sub-process path for both, then I'm game to upgrade to full sub-process. Though perhaps as a next step? There will be at least some minor refactoring necessary because the in-process vs. sub-process paths look like they operate on different types.

Reply:
Agreed, it's not necessary to do immediately.